src/hubmap_translation/addl_index_transformations/portal/add_assay_details.py

```diff
@@ -33,6 +48,58 @@ def _get_assay_details(doc, transformation_resources):
         raise
 
 
+def _add_dataset_processing_fields(doc):
+    if processing_type := processing_type_map.get(doc['creation_action']):
+        doc['processing'] = 'processed'
+        doc['processing_type'] = processing_type
+    else:
+        doc['processing'] = 'raw'
+
+
+def _is_component_dataset(doc):
+    if 'creation_action' in doc and doc['creation_action'] == CreationAction.MULTI_ASSAY_SPLIT:
+        return True
+    return False
+
+
+def _add_multi_assay_fields(doc, assay_details):
+    if _is_component_dataset(doc):
+        doc['assay_modality'] = 'multiple'
+        doc['multi_assay_category'] = 'component'
+        return
+
+    if assay_details.get('is-multi-assay', False):
+        doc['assay_modality'] = 'multiple'
+        creation_action = doc.get('creation_action', None)
+        if creation_action == CreationAction.CREATE_DATASET:
+            doc['multi_assay_category'] = 'primary'
+        elif creation_action == CreationAction.CENTRAL_PROCESS:
+            doc['multi_assay_category'] = 'processed'
```

Comment:
Feels a bit in conflict with the values in `processing`. Could this just be a `is_component` bool?

Reply:
I share your concern here, especially as it creates the potential for a conflict with `processing`. Let's change this to `is_component`.